### PR TITLE
Lift IR abstraction level with comparator and sort operations

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
The Rust code generation has been pushed in the visitor.
The high-level nodes should be more amenable for cross-platform code generation (hopefully for the JIT compiler too).

### CODE REVIEW

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java [hunk 1]
```diff
@@ -998,49 +996,27 @@ public void visitSort(LogicalSort sort) {
...
                 case STRICTLY_ASCENDING:
                 case STRICTLY_DESCENDING:
                 case CLUSTERED:
                     throw new Unimplemented(sort);
             }
-            if (comparators == null)
-                comparators = comparator;
-            else
-                comparators = new DBSPApplyMethodExpression("then", DBSPTypeAny.instance, comparators, comparator);
+            comparator = new DBSPFieldComparatorExpression(sort, comparator, field, ascending);
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:1017]: Why not just send the enum through rather a bool ascending=true/false? It's more informative than a boolean argument anyway.

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java [hunk 2]
```diff
@@ -998,49 +996,27 @@ public void visitSort(LogicalSort sort) {
...
+            boolean ascending;
             switch (direction) {
                 case ASCENDING:
+                    ascending = true;
                     break;
                 case DESCENDING:
-                    comparator = new DBSPApplyMethodExpression("rev", DBSPTypeAny.instance, comparator);
+                    ascending = false;
                     break;
+                default:
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:1011]: Where do these sort orders come from? An order by? From the sound of it it looks like a column definition?

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java (hunk 1 above)
Comment (mbudiu-vmw) [reply SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:1017]: In principle the IR classes are independent of the Sql front-end classes, so they could work for other front-ends.
I also don't know the meaning of the other enumeration values in this case.

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java (hunk 2 above)
Comment (mbudiu-vmw) [reply SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java:1011]: This is in the function visitSort, which implements the OrderBy clause.

**Comment by lalithsuresh** [comment]
LGTM

**Pull request merged**